src/hotspot/cpu/x86/x86.ad

```diff
@@ -1868,10 +1868,11 @@ const bool Matcher::match_rule_supported_vector(int opcode, int vlen, BasicType
       }
       break;
     case Op_VectorCastD2X:
-      if (is_subword_type(bt) || bt == T_INT) {
+      if (((UseAVX <= 2) || (!VM_Version::supports_avx512vl())) &&
```

Comment:
Which asm instructions are required avx512vl? I don't see asserts in `assembler_x86.cpp`

Reply:
avx512vl support is needed only for vectors < 512 bit. I have corrected this in the predicate.